build/gulp/generator.js

```diff
@@ -0,0 +1,44 @@
+const gulp = require('gulp');
+const { generateComponents } = require('devextreme-generator/component-compiler');
+const generator = require('devextreme-generator/preact-generator').default;
+const ts = require('gulp-typescript');
+const lint = require('gulp-eslint');
+const plumber = require('gulp-plumber');
+
+const SRC = 'js/**/*.tsx';
+const DEST = 'js';
+
+const knownErrors = [
+    'Cannot find module \'preact\'.',
+    'Cannot find module \'preact/hooks\'.'
+];
+
+gulp.task('generate-components', function() {
+    const tsProject = ts.createProject('build/gulp/preact.tsconfig.json');
+    return gulp.src(SRC)
+        .pipe(generateComponents(generator))
+        .pipe(plumber({
+            errorHandler: function() {
+
```

Comment:
Where is a function body? :)

Reply:
We do not need it. The plumber is used to preventing crashing Gulp Watch. we handle errors bellow in the tsProject reporter

Reply:
ok